File: iOS/EvianBlog/api/writings/TagAPI.swift
//
//  TagAPI.swift
//  EvianBlog
//
//  Created by Evian张 on 2020/5/18.
//  Copyright © 2020 Evian张. All rights reserved.
//

import Alamofire

import Foundation
import Combine

extension BlogAPI {
	func getTags() -> AnyPublisher<[Tag], BlogAPIError> {
		self.fetch(url: Endpoint.Writings.Tag.getTags)
	}
	
	func getArticlesCountOfTag(name: String) -> AnyPublisher<UInt, BlogAPIError> {
		self.fetch(url: Endpoint.Writings.Tag.getArticlesCountOfTag(name))
	}
	
	func getArticlesOfTag(name: String, pageIndex: UInt, pageSize: UInt) -> AnyPublisher<[ArticleMeta], BlogAPIError> {
		self.fetch(url: Endpoint.Writings.Tag.getArticlesOfTag(name, pageIndex, pageSize))
	}
}

File: Android/app/src/main/java/top/evian_zhang/evianblog/api/writings/TagAPI.kt
package top.evian_zhang.evianblog.api.writings

import top.evian_zhang.evianblog.api.ArticleMeta
import top.evian_zhang.evianblog.api.Tag
import top.evian_zhang.evianblog.api.BlogAPI
import top.evian_zhang.evianblog.api.Endpoint

suspend fun BlogAPI.getTags(): List<Tag> {
    return this.fetch(Endpoint.Writings.Tag.getTags)
}

suspend fun BlogAPI.getArticlesCountOfTag(name: String): Int {
    return this.fetch(Endpoint.Writings.Tag.getArticlesCountOfTag(name))
}

suspend fun BlogAPI.getArticlesOfTag(name: String, pageIndex: Int, pageSize: Int): List<ArticleMeta> {
    return this.fetch(Endpoint.Writings.Tag.getArticlesOfTag(name, pageIndex, pageSize))
}
